server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -188,19 +189,40 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
             Names.GENERIC,
             new ScalingExecutorBuilder(Names.GENERIC, 4, genericThreadPoolMax, TimeValue.timeValueSeconds(30), false)
         );
-        builders.put(Names.WRITE, new FixedExecutorBuilder(settings, Names.WRITE, allocatedProcessors, 10000, true));
-        builders.put(Names.GET, new FixedExecutorBuilder(settings, Names.GET, searchOrGetThreadPoolSize(allocatedProcessors), 1000, false));
-        builders.put(Names.ANALYZE, new FixedExecutorBuilder(settings, Names.ANALYZE, 1, 16, false));
+        builders.put(Names.WRITE, new FixedExecutorBuilder(settings, Names.WRITE, allocatedProcessors, 10000, new TaskTrackingConfig(0.1)));
```

Comment:
Here and in other places in the PR, I'm not at all sure what's 0.1 compared to the previous 0.3. Has this been already discussed/agreed? I'm starting to think I may not be the best one to review this finally.

Reply:
The new value is based on some estimate that is mentioned in the attached Jira issue if you are interested. We could change the value to something else later, but for now it seems the default (0.3) might not be ideal. At the end the main change is making it configurable, the value itself is only having some effect on what we consider is the average task length for autoscaling estimates.

Reply:
I see thanks! Makes sense by reading it quickly, but I'll leave it to one of the autoscaling colleagues to finally approve the 0.3->0.1 value in these places cc @idegtiarenko . All the rest of the PR LGTM!